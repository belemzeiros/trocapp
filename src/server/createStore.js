import { createStore, applyMiddleware } from 'redux';
import { combineReducers } from 'redux-immutable';
import { fromJS } from 'immutable';
import createSaga, { END } from 'redux-saga';

import reducers from '@/api/reducers';
import rootSaga from '@/api/sagas';

export const runAllSagas = store => {
  store.dispatch(END);
  return store.saga;
};

export default () => {
  const saga = createSaga();
  const store = createStore(
    combineReducers(reducers),
    fromJS({}),
    applyMiddleware(saga)
  );

  store.saga = saga.run(rootSaga);

  return store;
};
