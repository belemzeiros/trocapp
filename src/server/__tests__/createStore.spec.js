import { END } from 'redux-saga';
import createStore, { runAllSagas } from '../createStore';

describe('createStore', () => {
  it("should call redux's createStore", () => {
    const store = createStore();
    expect(store).toMatchSnapshot();
  });

  it('should call runAllSagas', () => {
    const store = {
      dispatch: jest.fn(),
      saga: jest.fn(),
    };
    const result = runAllSagas(store);
    expect(store.dispatch).toBeCalledWith(END);
    expect(result).toEqual(store.saga);
  });
});
