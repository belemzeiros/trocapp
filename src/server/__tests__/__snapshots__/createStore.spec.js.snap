// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createStore should call redux's createStore 1`] = `
Object {
  "dispatch": [Function],
  "getState": [Function],
  "replaceReducer": [Function],
  "saga": Object {
    "@@redux-saga/TASK": true,
    "cancel": [Function],
    "cont": undefined,
    "done": Promise {},
    "error": [Function],
    "id": 1,
    "isAborted": [Function],
    "isCancelled": [Function],
    "isRunning": [Function],
    "joiners": null,
    "name": "rootSaga",
    "result": [Function],
    "setContext": [Function],
  },
  "subscribe": [Function],
  Symbol(Symbol.observable): [Function],
}
`;
